### PR TITLE
oci: Client improvements

### PR DESCRIPTION
**Pull request by lann**
- Make Client::pull not silently ignore some IO errors
- De-nest code in a few places

### CODE REVIEW

File: crates/oci/src/client.rs
Comment (lann) [inline crates/oci/src/client.rs:1]: This is a bit easier to read with "Hide whitespace"
2023-10-18T16:10:35,697240244-04:00

Comment (lann) [reply crates/oci/src/client.rs:1]: Nope I lied that is awful.

File: crates/oci/src/client.rs [hunk 1]
```diff
@@ -195,18 +194,19 @@ impl Client {
...
-                // Add content/path to the locked component files list
-                let layer = Self::data_layer(entry.path(), DATA_MEDIATYPE.to_string()).await?;
-                let content = Self::content_ref_for_layer(&layer);
-                files.push(ContentPath {
-                    content,
-                    path: rel_path.into(),
-                });
+            if !entry.file_type().is_file() {
+                continue;
             }
```
Comment (lann) [inline crates/oci/src/client.rs:199]: This is the actual change here; the rest is just indent.

File: crates/oci/src/client.rs [hunk 2]
```diff
@@ -235,25 +235,26 @@ impl Client {
...
-                // As a workaround for OCI implementations that don't support very small blobs,
-                // don't push very small content that has been inlined into the manifest:
-                // https://github.com/distribution/distribution/discussions/4029
-                let skip_layer = content_inline;
-                if !skip_layer {
-                    layers.push(layer);
-                }
+            if !entry.file_type().is_file() {
+                continue;
+            }
```
Comment (lann) [inline crates/oci/src/client.rs:240]: Ditto re indent.

File: crates/oci/src/client.rs [hunk 3]
```diff
@@ -297,32 +298,23 @@ impl Client {
...
-                                        return Err(OciDistributionError::GenericError(Some(
-                                            e.to_string(),
-                                        )));
-                                    }
-                                }
-                                _ => {
-                                    let _ = this.cache.write_data(&bytes, &layer.digest).await;
-                                }
-                            },
+                        return anyhow::Ok(());
```
[the remark this answers is not in the document]
Comment (vdice) [reply crates/oci/src/client.rs:301]: Nice!

**Review by vdice — APPROVED** [review]

**Pull request merged**